pkg/k8s/watchers/cilium_endpoint.go

```diff
@@ -108,7 +111,19 @@ func (k *K8sWatcher) endpointUpdated(oldEndpoint, endpoint *types.CiliumEndpoint
 			k.policyManager.TriggerPolicyUpdates(true, "Named ports added or updated")
 		}
 	}()
-
+	// handle endpoint_propagation_delay_seconds Metric for CEP create
+	if oldEndpoint == nil && endpoint != nil {
+		if p := k.endpointManager.LookupPodName(k8sUtils.GetObjNamespaceName(endpoint)); p != nil {
+			timeSinceCepCreated := time.Since(p.GetCreatedAt())
+			metrics.EndpointPropagationDelay.WithLabelValues().Observe(timeSinceCepCreated.Seconds())
+		}
+		// handle endpoint_propagation_delay_seconds Metric for CEP update
+	} else if endpoint != nil && oldEndpoint != nil {
+		if p := k.endpointManager.LookupPodName(k8sUtils.GetObjNamespaceName(oldEndpoint)); p != nil {
```

Comment:
I have a feeling this name based lookup will just return the latest endpoint, what do you think ?

Reply:
thanks for your patience:), i got it

Reply:
@sayboras , update operation has some different traits with creation, we can determine endpoint creation time, but every update will start an endpoint modify time, maybe should using an revision mechanism like https://github.com/cilium/cilium/pull/7160? set a callback for this metric? i will take a try